File: backend/app/models/finance_qa.py
from pydantic import BaseModel, Field


class FinanceQuestionRequest(BaseModel):
    question: str = Field(min_length=2)


class FinanceCitation(BaseModel):
    title: str
    authority: str
    document_number: str | None = None
    published_date: str
    status: str
    source_url: str
    updated_at: str


class FinanceQuestionResponse(BaseModel):
    question: str
    intent: str
    answer: str
    confidence: float = Field(ge=0, le=1)
    action_items: list[str]
    citations: list[FinanceCitation]
    risk_level: str
    requires_human_review: bool = True
    latest_policy_check_required: bool = True

File: frontend/src/types/financeQa.ts
export interface FinanceQuestionRequest {
  question: string;
}

export interface FinanceCitation {
  title: string;
  authority: string;
  document_number: string | null;
  published_date: string;
  status: string;
  source_url: string;
  updated_at: string;
}

export interface FinanceQuestionResponse {
  question: string;
  intent: string;
  answer: string;
  confidence: number;
  action_items: string[];
  citations: FinanceCitation[];
  risk_level: "low" | "medium" | "high" | string;
  requires_human_review: boolean;
  latest_policy_check_required: boolean;
}
